validations/src/validations.rs

```diff
@@ -1326,6 +1328,7 @@ pub fn validate_block_transactions(
     block_number: u32,
     consensus_constants: &ConsensusConstants,
     active_wips: &ActiveWips,
+    visitor: &mut Option<&mut dyn Visitor<Visitable = (Transaction, u64, u32), State = T>>,
```

Comment:
`&mut Option<&mut T>` is never what you want, because with a `Option<&mut T>` you can already modify `T`.

```suggestion
    mut visitor: Option<&mut dyn Visitor<Visitable = (Transaction, u64, u32), State = T>>,
```

That gives some compile errors because you will need to match on `&mut visitor`:

```
        if let Some(visitor) = &mut visitor {
```

Reply:
How can I solve the compile errors? I put the `&mut`up front because that was the only way I got it to compile  :cry: